optimade/adapters/structures/aiida.py

```diff
@@ -60,12 +62,21 @@ def get_aiida_structure_data(optimade_structure: OptimadeStructure) -> Structure
                 symbols.append(chemical_symbol)
                 concentration.append(kind.concentration[index])
 
-        # AiiDA needs a definition for the mass, and for it to be > 0
-        # mass is OPTIONAL for OPTIMADE structures
-        mass = kind.mass if kind.mass else 1
+                # AiiDA needs a definition for the mass, and for it to be > 0
+                # mass is OPTIONAL for OPTIMADE structures
+                if kind.mass:
+                    mass += kind.concentration[index] * kind.mass[index]
+
+        if not mass:
+            warn(
+                f"No mass defined for <species(name={kind.name!r})>, will default to setting mass to 1.0.",
+                ConversionWarning,
+            )
```

Comment:
I think my suggestion was more along the lines of:

```suggestion
        if kind.mass is not None and len(kind.mass) > 1:
            warn(
                f"Custom mass defined in structure for <species(name={kind.name!r}>, will default to setting mass to 1.0.",
                ConversionWarning,
            )
```

i.e. just disabling this kind of handling. I'm happy to go with this if you think it is robust enough though.

Reply:
No, since mass has gone from being a sum to being a list of the parts, the mass must now be calculated to the sum again in order to be used in AiiDA. Hence, if the mass is not defined (it will be `0.0`) and simply set to `1.0` for the `Kind` initialization.